server/db/driver/pg/pg.go

```diff
@@ -80,6 +83,18 @@ type Archiver struct {
 	tables        archiverTables
 }
 
+// LastErr returns any fatal or unexpected error encountered in a recent query.
+// This may be used to check if the database had an unrecoverable error
+// (disconnect, etc.).
+func (a *Archiver) LastErr() error {
+	err, _ := a.lastErr.Load().(error)
+	return err
+}
+
+func (a *Archiver) unexpectedErr(err error) {
```

Comment:
Not crazy about the name. Virtually all errors are unexpected in some context.

Reply:
`fatalBackendErr`?